### PR TITLE
add eslint:recommended and plugin:@typescript-eslint/recommended

### PR DESCRIPTION
**Pull request by miettal**
eslint:recommended and  plugin:@typescript-eslint/recommended commonly used in eslint extends in typescript application development.

This changes add these extends to openems-ui project.

// Some rules was disabled because there are many errors, but in future resolving is better.

### CODE REVIEW

File: ui/src/app/edge/history/storage/chargerchart.component.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export class StorageChargerChartComponent extends AbstractHistoryChart implement
 
     ngOnChanges() {
         this.updateChart();
-    };
+    }
```
Comment (lukasrgr) [inline ui/src/app/edge/history/storage/chargerchart.component.ts:22]: this change is weird

File: ui/src/app/edge/live/Controller/Io/ChannelSingleThreshold/modal/modal.component.ts [hunk 2]
```diff
@@ -91,7 +91,6 @@ export class Controller_Io_ChannelSingleThresholdModalComponent implements OnIni
           this.threshold.setValue(newThreshold);
           this.threshold.markAsDirty();
         } else if (this.component.properties.threshold < 0) {
-          newThreshold = newThreshold;
```
Comment (lukasrgr) [inline ui/src/app/edge/live/Controller/Io/ChannelSingleThreshold/modal/modal.component.ts:94]: filtering something like this out, is very good

File: ui/src/app/edge/live/Multiple/Evcs_Api_Cluster/modal/evcs-chart/evcs.chart.ts [hunk 3]
```diff
@@ -70,19 +70,18 @@ export class EvcsChartComponent implements OnInit, OnChanges {
...
-    let maxPower: number;
-    let minPower = 22;
+    const minPower = 22;
     let maxHW = this.currentData[this.componentId + '/MaximumHardwarePower'];
     let chargePower = this.currentData[this.componentId + '/ChargePower'];
     maxHW = maxHW == null ? minPower : maxHW / 1000;
     chargePower = chargePower == null ? 0 : chargePower / 1000;
 
-    maxPower = chargePower < minPower || maxPower < minPower ? minPower : maxHW;
+    const maxPower: number = chargePower < minPower || maxHW;
```
Comment (lukasrgr) [inline ui/src/app/edge/live/Multiple/Evcs_Api_Cluster/modal/evcs-chart/evcs.chart.ts:84]: working great 👍

File: ui/src/app/edge/history/storage/chargerchart.component.ts (hunk 1 above)
Comment (lukasrgr) [reply ui/src/app/edge/history/storage/chargerchart.component.ts:22]: having no semi-colons at the end of function body is recommended??

Comment (miettal) [reply ui/src/app/edge/history/storage/chargerchart.component.ts:22]: in JavaScript syntax, Function Definetion is not included semicoron.
This semicoron is a Empty Statement.
Perhaps This semi is typo.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/Empty

File: ui/src/app/edge/live/Multiple/Evcs_Api_Cluster/modal/evcs-chart/evcs.chart.ts (hunk 3 above)
Comment (miettal) [reply ui/src/app/edge/live/Multiple/Evcs_Api_Cluster/modal/evcs-chart/evcs.chart.ts:84]: memo: for easy understanding

original sentense use `maxPower` without initialization so it assumed as `undefined`.
so the sentence will evaluated following steps

1. maxPower = chargePower < minPower || maxPower < minPower ? minPower : maxHW;
2. maxPower = chargePower < minPower || undefined < minPower ? minPower : maxHW;
3. maxPower = chargePower < minPower || false ? minPower : maxHW;
4. maxPower = chargePower < minPower || maxHW;

so I rewrited by final sentense.

**Comment by sfeilmeier** [comment]
Thank you @miettal. The PR is ready to get merged after Lukas' review. Unfortunately it has conflicts with develop and I don't have write permissions on the branch. Could you please either merge from develop or give me write access to fix it myself? Thanks!

**Comment by miettal** [comment]
OK. I understood the situation. I will rebase this PR.

**Comment by miettal** [comment]
> I will rebase this PR.

done

**Pull request merged**